src/screens/SpeedUpAndCancelSheet.js

```diff
@@ -145,23 +157,57 @@ export default function SpeedUpAndCancelSheet() {
   const [to, setTo] = useState(tx.to);
   const [value, setValue] = useState(null);
 
-  const getNewGasPrice = useCallback(() => {
-    const rawGasPrice = get(selectedGasPrice, 'value.amount');
-    const minGasPriceAllowed = gweiToWei(minGasPrice);
-    const rawGasPriceBN = new BigNumber(rawGasPrice);
-    const minGasPriceAllowedBN = new BigNumber(minGasPriceAllowed);
-    return rawGasPriceBN.isGreaterThan(minGasPriceAllowedBN)
-      ? toHex(rawGasPrice)
-      : toHex(minGasPriceAllowed);
-  }, [minGasPrice, selectedGasPrice]);
+  const getNewTransactionGasParams = useCallback(() => {
+    if (txType === EIP1559TransactionTypes.current) {
+      const rawMaxPriorityFeePerGas =
+        selectedGasFee?.gasFeeParams?.maxPriorityFeePerGas?.amount;
+      const rawMaxFeePerGas =
+        selectedGasFee?.gasFeeParams?.maxFeePerGas?.amount;
+
+      const rawMaxPriorityFeePerGasBN = new BigNumber(rawMaxPriorityFeePerGas);
+      const minMaxPriorityFeePerGasAllowedBN = new BigNumber(
+        minMaxPriorityFeePerGas
+      );
+      const rawMaxFeePerGasBN = new BigNumber(rawMaxFeePerGas);
+      const minMaxFeePerGasAllowedBN = new BigNumber(minMaxFeePerGas);
+
+      const maxPriorityFeePerGas = rawMaxPriorityFeePerGasBN.isGreaterThan(
```

Comment:
all these BN transformations can be replaced with `greaterThan` from `utilities`

Reply:
👌 👌 👌 👌